spec/nio/monitor_spec.rb

```diff
@@ -51,6 +57,25 @@
     expect(reader_monitor).not_to be_writable
   end
 
+  it "Changes the interest_set on the go" do
+    # Only works in CRuby for some reason.
+    # As I identified it might be because of the differences between
+    # two implementations (JRuby and CRuby)
+    # In Jruby we cannot even use the (:W) to register a ReaderMonitor
+    # (selector.register(reader, :r)) because of "IllegalArgumentException"
```

Comment:
Can you change `subject` and `peer` to be a `TCPServer` and `TCPSocket`? I think that will cover what we actually care about and make things work more consistently between CRuby and JRuby. Pipes are weird and Java doesn't really expose them well.